### DISCUSSION TITLE
centeredSlides: 'true' and .swiper-slide { width: auto; } make the slider start at the last image with missing images to the right

### DISCUSSION THREAD

**Discussion by Stanzi1791** [nolimits4web/swiper#5075] [Q&A] [answered]
Hi,

When I'm using centeredSlides: 'true' in combination with .swiper-slide { width: auto; }
the slider starts with the last image centered, images to the right of the centered image are missing.

<img src="https://camo.githubusercontent.com/f530af0312b56dd678e7fcc2737fb1d9b2493c139edd271d608094868f0c4daa/68747470733a2f2f692e6962622e636f2f6b5358517774592f53637265656e73686f742d323032312d31302d31302d3132353731372e706e67" alt="Screenshot-2021-10-10-125717" border="0" data-canonical-src="https://i.ibb.co/kSXQwtY/Screenshot-2021-10-10-125717.png" style="max-width: 100%;">

After the first autoplay "move", the first image is centered and the images to the right are loaded,
this is the state I would have expected Swiper to start in.

<img src="https://camo.githubusercontent.com/d2eff1032326bd738d8f76c4dd71ae8604cfa145ab0b6d53bba2b366b7441f6f/68747470733a2f2f692e6962622e636f2f4e73794e6b32622f53637265656e73686f742d323032312d31302d31302d3132353732332e706e67" alt="Screenshot-2021-10-10-125723" border="0" data-canonical-src="https://i.ibb.co/NsyNk2b/Screenshot-2021-10-10-125723.png" style="max-width: 100%;">

Fiddle here (the first image is the green one):
https://jsfiddle.net/wq7j8Lep/2/ (you may have to refresh the page to see the behaviour above)

The fiddle uses Swiper 7.0.8, but I've also seen this behaviour in Swiper version 5.3.6.

If I turn off autoplay and look at the html, it looks like the "swiper-slide-active" class is getting assigned to the wrong slide (the last one) when the page first loads. Did I use the plugin wrong or is there a workaround for this?

Kind regards,
Marleen

**Comment by Stanzi1791** [comment, marked as answer]
I went through the Swiper code and noticed the for loop inside the `if (params.normalizeSlideIndex) { }` in the slideTo function is in this case incorrectly recalculating the slideIndex, causing the "swiper-slide-active" class to be added to the wrong slide.

Adding `normalizeSlideIndex: false` to the parameters of the Swiper appears to fix the problem.
It's not clear from the documentation what this parameter is supposed to do, so this might have other side effects I haven't noticed yet.

Hopefully this will be of use to others.

Fiddle with the fix: https://jsfiddle.net/wxo9ugkz/

**Reply by myrecs** [reply]
also faced with problem that active index calculated wrong in case when using Auto slides per view and slideToClickedSlide:true. It behaves like so: if left slide if hidden less than half of its width (in code it calculates width / 2), then swiper doesnt slide to it, because it still sets activeIndex to that left slide, but it should slide, cause only part of slide is visible! Disabling normalizeSlideIndex solves this issue, but i think it's a bug and you should add some cases in code for such situations.
